plugin/trace/setup.go

```diff
@@ -34,14 +34,30 @@ func setup(c *caddy.Controller) error {
 	return nil
 }
 
+func endPointName(hosts []string, port string) string {
+	host := ""
+	switch len(hosts) {
+	case 0:
```

Comment:
can this happen?

Reply:
no more.

Reply:
although .. I do not like to make a len(xx) without testing the zero-length.